lib/chef/resource/launchd.rb

```diff
@@ -46,6 +46,51 @@ def initialize(name, run_context = nil)
       property :source, String
       property :session_type, String
 
+      # StartCalendarInterval has some gotchas so we coerce it to help sanity
+      # check.  According to `man 5 launchd.plist`:
+      #   StartCalendarInterval <dictionary of integers or array of dictionaries of integers>
+      #     ... Missing arguments are considered to be wildcard.
+      # What the man page doesn't state, but what was observed (OSX 10.11.5, launchctrl v3.4.0)
+      # Is that keys that are specified, but invalid, will also be treated as a wildcard
+      # this means that an entry like:
+      #   { "Hour"=>0, "Weekday"=>"6-7"}
+      # will not just run on midnight of Sat and Sun, rather it will run _every_ midnight.
+      property :start_calendar_interval, [Hash, Array], coerce: proc { |type|
+        # Coerce into an array of hashes to make validation easier
+        array = if type.is_a?(Array)
+                  type
+                else
+                  [type]
+                end
+
+        # Check to make sure that our array only has hashes
+        if array.any? { |obj| !obj.is_a?(Hash) }
```

Comment:
negating the condition actually reads better to me:

`unless array.all? { |obj| obj.is_a?(Hash) }`